modules/field_event_dispatcher/tests/src/Unit/Field/FormEntityViewDisplayEditAlterEventSubscriberTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Drupal\Tests\field_event_dispatcher\Unit\Field;
+
+use Drupal;
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\core_event_dispatcher\Event\Form\FormIdAlterEvent;
+use Drupal\field_event_dispatcher\EventSubscriber\Form\FormEntityViewDisplayEditAlterEventSubscriber;
+use Drupal\hook_event_dispatcher\HookEventDispatcherInterface;
+use Drupal\Tests\hook_event_dispatcher\Unit\HookEventDispatcherManagerSpy;
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * Class FormEntityViewDisplayEditAlterEventSubscriberTest.
+ *
+ * @group field_event_dispatcher
+ */
+class FormEntityViewDisplayEditAlterEventSubscriberTest extends UnitTestCase {
+
+  /**
+   * The manager.
+   *
+   * @var \Drupal\Tests\hook_event_dispatcher\Unit\HookEventDispatcherManagerSpy
+   */
+  private $manager;
+
+  /**
+   * Sets up the test.
+   */
+  public function setUp(): void {
+    $builder = new ContainerBuilder();
+    $this->manager = new HookEventDispatcherManagerSpy();
+    $builder->set('hook_event_dispatcher.manager', $this->manager);
+    $builder->compile();
+    Drupal::setContainer($builder);
+  }
+
+  /**
+   * FormEntityViewDisplayEditAlterEventSubscriber form alter test.
+   *
+   * This tests that FormEntityViewDisplayEditAlterEventSubscriber::formAlter()
+   * alters the $form array as expected.
+   */
+  public function testFormAlter(): void {
+    $form = [
+      '#fields' => ['field_test'],
+    ];
+    $formId = 'entity_view_display_edit_form';
+
+    $form['fields']['field_test']['plugin']['settings_edit_form']['third_party_settings']['field_event_dispatcher']['test_module'] = 'test';
+
+    $expectedForm = $form;
+
+    $eventSubscriber = new FormEntityViewDisplayEditAlterEventSubscriber();
+    $formState = $this->createMock(FormStateInterface::class);
+    $event = new FormIdAlterEvent($form, $formState, $formId);
+
+    $eventSubscriber->formAlter($event);
+
+    $expectedForm['fields']['field_test']['plugin']['settings_edit_form']['third_party_settings']['test_module'] = $expectedForm['fields']['field_test']['plugin']['settings_edit_form']['third_party_settings']['field_event_dispatcher']['test_module'];
+
+    unset($expectedForm['fields']['field_test']['plugin']['settings_edit_form']['third_party_settings']['field_event_dispatcher']);
+
+    $this->assertEquals($expectedForm, $event->getForm());
```

Comment:
`assertSame` can be used to strict check the two arrays.

Reply:
Will do!